### PR TITLE
feat(weblate/settings_docker.py): Add USE_X_FORWARDED_HOST key in settings_docker.py file.

### LINKED ISSUE #8566: Allow setting Django USE_X_FORWARDED_HOST in `settings_docker.py`
### Describe the problem

It is documented [here](https://docs.weblate.org/en/latest/admin/install.html#running-behind-reverse-proxy) that:
> Additional configuration might be needed in your reverse proxy (for example use ProxyPreserveHost On for Apache or proxy_set_header Host $host; with nginx).

Altought this is a solution, it is sometimes difficult to adjust the proxy settings.

### Describe the solution you'd like

It would be nicer to be able to configure Weblate to use an "X-Forwarded-Host" header which is often supplied.
This is supported by Django with the [USE_X_FORWARDED_HOST](https://docs.djangoproject.com/en/4.1/ref/settings/#use-x-forwarded-host) setting.
This setting isn't currently exposed in the `settings_docker.py` so it can't be set with environment variables.

### Describe alternatives you've considered

Building a custom docker image with a custom `settings.py` which exposes the setting.
Or alternatively mounting a custom `settings.py` into the running container to expose it.

### Screenshots

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Pavankumardontha**
Fixes #8566

## Proposed changes
Added USE_X_FORWARDED_HOST key

## Checklist

- [ ] Lint and unit tests pass locally with my changes.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added documentation to describe my feature.
- [ ] I have squashed my commits into logic units.
- [ ] I have described the changes in the commit messages.

### CODE REVIEW

**Comment by nijel** [comment]
Can you please add documentation as well?

**Comment by Pavankumardontha** [comment]
@nijel can you please tell me how to add documentation ?? I am a newbie for open source

**Comment by nijel** [comment]
The Docker image configuration is described here: https://github.com/WeblateOrg/weblate/blob/main/docs/admin/install/docker.rst

**Comment by Pavankumardontha** [comment]
@nijel i have added the documentation part. Can you once please go through it and let me know if you need any more changes from my end ?? Thanks

File: docs/admin/install/docker.rst [hunk 1]
```diff
@@ -680,6 +680,17 @@ Generic settings
         environment:
           WEBLATE_IP_PROXY_HEADER: HTTP_X_FORWARDED_FOR
 
+.. envvar:: WEBLATE_USE_X_FORWARDED_PORT
+
```
Comment (nijel) [inline docs/admin/install/docker.rst:684]:
```suggestion

    .. versionadded:: 5.0.1

```

File: docs/admin/install/docker.rst [hunk 2]
```diff
@@ -680,6 +680,17 @@ Generic settings
...
+    A boolean that specifies whether to use the X-Forwarded-Port header in
+    preference to the SERVER_PORT META variable. This should only be enabled
+    if a proxy which sets this header is in use.
+
+    .. seealso::
+
+        :setting:`django:USE_X_FORWARDED_PORT`
+
+    .. note:: This is a Boolean setting (use ``"true"`` or ``"false"``).
```
Comment (nijel) [inline docs/admin/install/docker.rst:693]:
```suggestion
    .. note:: This is a boolean setting (use ``"true"`` or ``"false"``).
```

File: docs/admin/install/docker.rst [hunk 3]
```diff
@@ -680,6 +680,17 @@ Generic settings
         environment:
           WEBLATE_IP_PROXY_HEADER: HTTP_X_FORWARDED_FOR
 
+.. envvar:: WEBLATE_USE_X_FORWARDED_PORT
+
+    A boolean that specifies whether to use the X-Forwarded-Port header in
```
Comment (nijel) [inline docs/admin/install/docker.rst:685]:
```suggestion
    A boolean that specifies whether to use the :http:header:`X-Forwarded-Port` header in
```

**Comment by nijel** [comment]
Merged, thanks for your contribution!

**Comment by Pavankumardontha** [comment]
@nijel  Thanks for this. !! Would surely like to contribute more for in this project.

**Pull request merged**